native/src/snappyclient/cpp/types/Decimal.cpp

```diff
@@ -360,15 +360,16 @@ size_t Decimal::toString(std::string& str) const {
   char buf[thrift::snappydataConstants::DECIMAL_MAX_PRECISION + 4];
   char* bufp = buf;
   io::snappydata::client::impl::FreePointer freep(0);
-  const size_t ndigits = mpz_sizeinbase(m_bigInt, 10);
+  size_t ndigits = mpz_sizeinbase(m_bigInt, 10);
   if (ndigits > 128) {
     bufp = new char[ndigits + 2];
     freep.reset(bufp);
   }
   mpz_get_str(bufp, 10, m_bigInt);
-
+  ndigits = strlen(bufp);// to get exact size
```

Comment:
Comment seems to be misleading. Should be corrected.

Reply:
Moved the comment on its right place